server/lib/pbench/process_tb.py

```diff
@@ -0,0 +1,108 @@
+import os
+import glob
+import subprocess
+
+from collections import namedtuple
+from logging import Logger
+from pathlib import Path
+
+from pbench import PbenchConfig
+
+Results = namedtuple("Results", ["nstatus", "ntotal", "ntbs"])
+
+
+class ProcessTb:
+    def __init__(self, config: PbenchConfig, logger: Logger):
+        """ Processes Tar ball, gets the tar ball from receiving directory
+        and copies it to the remote server using HTTP PUT method.
+
+            Args -
+                config -- PbenchServer config object
+                logger -- logger object to use when emitting log entries during
+                            operation
+
+        """
+        self.config = config
+        self.logger = logger
+
+    def get_receive_dir(self) -> Path:
+        receive_dir_prefix = self.config.get(
+            "pbench-server", "pbench-receive-dir-prefix"
+        )
+        if not receive_dir_prefix:
+            self.logger.error(
+                "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
+            )
+            return None
+
+        receive_dir = Path(f"{receive_dir_prefix}-002").resolve()
+        if not receive_dir.is_dir():
+            self.logger.error(
+                "Failed: {} does not exist, or is not a directory", receive_dir
+            )
+            return None
+
+        return receive_dir
+
+    def run_subprocess(self, controller: str, tb: str, token: str):
+        """Runs `pbench-results-push` command with controller, tb
+        and token as options"""
+
+        subprocess.run(["pbench-results-push", controller, tb, token])
+
+    def process_tb(self, receive_dir: Path) -> Results:
```

Comment:
Then it will mean I should get `get_receive_dir` function out of `Process_Tb` class. If I want to provide `receive_dir` at the time of initialization than it should be already available to me.

Reply:
Yes, exactly.  So, the class interface will be smaller and the client side will be simpler.  The unit testing will be pretty much the same, except that the tests which used to test `get_receive_dir()` will now target the class constructor...and, it will need to report errors via exceptions, because the c'tor can't provide the return value.